### PR TITLE
Always name `variant` nodes even if there is only single primary expr

### LINKED ISSUE #802: PRATT parser doesn't insert correct field names on primary expression
Example in the `VersionPragmaSpecifier` node here, which should have had a `variant:` field name, but it doesn't:

https://github.com/NomicFoundation/slang/blob/29ae3137d98edeeb617fc621339f4698bc21b9e5/crates/solidity/testing/snapshots/cst_output/VersionPragma/equal_operator/generated/0.4.11-success.yml#L15C27-L15C50

### PR DESCRIPTION
**Pull request by Xanewok**
Closes #802

### CODE REVIEW

**Comment by Xanewok** [comment]
@OmarTawfik @AntonyBlakey  do we want to always name the node with variant, even though there is only a single primary expression?

**Comment by OmarTawfik** [comment]
> do we want to always name the node with variant, even though there is only a single primary expression?

even if there is a single `PrimaryExpression` variant, there are multiple `PrecedenceExpression` variants. They would all have `variant: ` field name. right?

File: crates/solidity/outputs/cargo/crate/src/generated/language.rs [hunk 1]
```diff
@@ -5605,8 +5605,14 @@ impl Language {
                 choice.finish(input)
             })
         };
-        let primary_expression_parser =
-            |input: &mut ParserContext<'_>| self.version_pragma_specifier(input);
+        let primary_expression_parser = |input: &mut ParserContext<'_>| {
```
Comment (OmarTawfik) [inline crates/solidity/outputs/cargo/crate/src/generated/language.rs:5608]: nit: shouldn't codegen remove the `ChoiceHelper` if it has only one element?

**Review by OmarTawfik — APPROVED** [review]

File: crates/solidity/outputs/cargo/crate/src/generated/language.rs (hunk 1 above)
Comment (Xanewok) [reply crates/solidity/outputs/cargo/crate/src/generated/language.rs:5608]: I'll follow up in a PR addressing #638, we need to clean this up a bit anyway

**Comment by Xanewok** [comment]
Although it's on the same level in the CST, it's not named as `variant` because it's not reduced in PrecedenceHelper (where we unconditionally name the sub-expressions as such), even though they may be in theory only a single PrecedenceExpression. As such, it makes sense to be consistent for PrimaryExpressions here; so I'll go ahead and merge this.

**Pull request merged**